esrally/mechanic/telemetry.py

```diff
@@ -790,40 +793,55 @@ def attach_to_node(self, node):
     def on_benchmark_start(self):
         if self.process is not None:
             self.process_start = sysstats.process_io_counters(self.process)
+            read_bytes = 0
+            write_bytes = 0
+            io.ensure_dir(self.log_root)
+            tmp_io_file = os.path.join(self.log_root, "%s.io" % self.node_name)
             if self.process_start:
+                read_bytes = self.process_start.read_bytes
+                write_bytes = self.process_start.write_bytes
                 self.logger.info("Using more accurate process-based I/O counters.")
             else:
                 try:
                     self.disk_start = sysstats.disk_io_counters()
+                    read_bytes = self.disk_start.read_bytes
+                    write_bytes = self.disk_start.write_bytes
                     self.logger.warning("Process I/O counters are not supported on this platform. Falling back to less accurate disk "
                                         "I/O counters.")
                 except RuntimeError:
                     self.logger.exception("Could not determine I/O stats at benchmark start.")
+            with open(tmp_io_file, "wt", encoding="utf-8") as f:
+                json.dump({"pid": self.node.pid, "read_bytes": read_bytes, "write_bytes": write_bytes}, f)
 
     def on_benchmark_stop(self):
-        if self.process is not None:
+            io.ensure_dir(self.log_root)
+            tmp_io_file = os.path.join(self.log_root, "%s.io" % self.node_name)
+            with open(tmp_io_file, "rt", encoding="utf-8") as f:
+                io_bytes = json.load(f)
```

Comment:
As you now also store the PID, the name `io_bytes` does not seem appropriate anymore? How about `io_stats`?